### DISCUSSION TITLE
Optional timeout on reads and writes

### DISCUSSION THREAD

**Discussion by stridge-cruxml** [esynr3z/corsair#102] [Questions]
Add an option to have a timeout on transactions. If read or write is not ready, it should timeout the transaction and return an error. Related to issue #8

**Comment by esynr3z** [comment]
Sounds nice. However, I don't really get what can be the source of such backpressure within generated map that timeouts CSR access?

Moreover, AFAIK timeout bridges or monitors are usually quite separate things from CSR maps.

Could you please elaborate your idea a bit or provide some practical examples?
